src/components/tab-title/tab-title.tsx

```diff
@@ -176,20 +222,42 @@ export class TabTitle implements InteractiveComponent {
         <div
           class={{
             container: true,
-            [CSS.iconPresent]: this.iconStart || this.iconEnd ? true : null,
-            [CSS.containerHasText]: this.hasText
+            [CSS.iconPresent]: !!this.iconStart || !!this.iconEnd
           }}
+          hidden={closed}
           // eslint-disable-next-line react/jsx-sort-props
           ref={(el) => this.resizeObserver?.observe(el)}
         >
-          {this.iconStart ? iconStartEl : null}
-          <slot />
-          {this.iconEnd ? iconEndEl : null}
+          <div class={{ [CSS.content]: true, [CSS.contentHasText]: this.hasText }}>
+            {this.iconStart ? iconStartEl : null}
+            <slot />
+            {this.iconEnd ? iconEndEl : null}
+          </div>
+          {this.renderCloseButtonEl()}
         </div>
       </Host>
     );
   }
 
+  renderCloseButtonEl(): VNode {
+    const { closable, messages } = this;
+
+    return closable ? (
+      <button
+        aria-label={messages.close}
+        class={CSS.closeButton}
+        disabled={false}
+        key={CSS.closeButton}
+        onClick={this.closeClickHandler}
+        onKeyDown={this.closeKeyHandler}
```

Comment:
is the keydown needed? I thought native button elements don't need keydown for enter/space support.